File: System/AppOther/SubUnit/src/main/java/com/appother/subunit/activity/WelcomeActivity.java
package com.appother.subunit.activity;

import android.os.Bundle;

import com.appother.subunit.welcome.style.MoveActionStyle;
import com.appother.subunit.welcome.style.StyleBase;
import com.yline.base.BaseActivity;

public class WelcomeActivity extends BaseActivity
{
	private StyleBase styleBase;

	private STYLE style = STYLE.move;

	@Override
	protected void onCreate(Bundle savedInstanceState)
	{
		super.onCreate(savedInstanceState);

		if (style == STYLE.move)
		{
			styleBase = new MoveActionStyle();
		}

		styleBase.init(this);
	}

	public enum STYLE
	{
		move;
	}
}

File: System/AppOther/Manager/src/main/java/com/manager/welcome/WelcomeActivity.kt
package com.manager.welcome

import android.content.Context
import android.content.Intent
import android.os.Bundle
import com.yline.base.BaseActivity

/**
 * 欢迎界面
 * @author yline 2018/6/5 -- 15:28
 * @version 1.0.0
 */
class WelcomeActivity : BaseActivity() {
    companion object {
        fun launcher(context: Context) {
            val intent = Intent()
            intent.setClass(context, WelcomeActivity::class.java)
            context.startActivity(intent)
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        WelcomeStyleManager.showMoveDownStyle(this@WelcomeActivity)
    }
}